cypress/integration/promisesPage.spec.js

```diff
@@ -0,0 +1,105 @@
+describe("Promises page", () => {
+  it("renders Promises page", () => {
+    cy.visit("/");
+    cy.get("[data-cy=navigation-buttons]").contains("Promises").click();
+    cy.location("pathname", { timeout: 10000 }).should("include", "/promises");
+    cy.get("[data-cy=promises-section]").contains("Promises");
+  });
+  it("renders Promises categories", () => {
+    cy.get("[data-cy=promises-section]").contains("Promises by status");
+    cy.get("[data-cy=promises-section]").contains("Promises by category");
+  });
+  it("filter in-progress promises", () => {
+    cy.get("[data-cy=promises-section]")
+      .find("img")
+      .its("length")
+      .then((length) => {
+        expect(length).to.equal(5);
+      });
+    cy.get("[data-cy=filter-buttons]").contains("In Progress").click();
+    cy.get("[data-cy=promises-section]")
+      .find("img")
+      .its("length")
+      .then((length) => {
+        expect(length).to.equal(3);
+      });
+    cy.get("[data-cy=filter-buttons]").contains("In Progress").click();
+  });
+
+  it("filter unrated promises", () => {
+    cy.get("[data-cy=promises-section]")
+      .find("img")
+      .its("length")
+      .then((length) => {
+        expect(length).to.equal(5);
+      });
+    cy.get("[data-cy=filter-buttons]").contains("Unrated").click();
+    cy.get("[data-cy=promises-section]")
+      .find("img")
+      .its("length")
+      .then((length) => {
+        expect(length).to.equal(1);
+      });
+    cy.get("[data-cy=filter-buttons]").contains("Unrated").click();
+  });
+
+  it("filter stalled promises", () => {
+    cy.get("[data-cy=promises-section]")
+      .find("img")
+      .its("length")
+      .then((length) => {
+        expect(length).to.equal(5);
+      });
```

Comment:
If I'm getting this correct, it means everytime the PromiseTracker team add/remove promises in Check, these tests will fail. Correct?

Reply:
Precisely. That's the problem I have at the moment, since I am using production database it is bound to change any time. And how to mock that data is a challenge on Cypress as it runs on the client side and the pages are SSR - so they already come with the data.